src/resolve/resolved-dns-packet.c

```diff
@@ -47,13 +47,7 @@ int dns_packet_new(DnsPacket **ret, DnsProtocol protocol, size_t mtu) {
 
         assert(ret);
 
-        if (mtu <= UDP_PACKET_HEADER_SIZE)
-                a = DNS_PACKET_SIZE_START;
-        else
-                a = mtu - UDP_PACKET_HEADER_SIZE;
-
-        if (a < DNS_PACKET_HEADER_SIZE)
-                a = DNS_PACKET_HEADER_SIZE;
+        a = MAX(mtu, DNS_PACKET_HEADER_SIZE);
```

Comment:
hmm, dns_packet_new() is sometimes called with mtu == 0, and in that case we should allocate more than the absolute minimum (which is the dns packet header size), otherwise we have to resize immediately again after appending the first data to the packet.

maybe something like this:

```
if (mtu < UDP_PACKET_HEADER_SIZE)
        mtu = DNS_PACKET_SIZE_START;

a = MAX(mtu, DNS_PACKET_HEADER_SIZE);
```
or so? that way if mtu is passed as 0, we will allocate DNS_PACKET_SIZE_START bytes, which is a good starting point.